lua/cmp/source.lua

```diff
@@ -115,7 +115,11 @@ source.get_entries = function(self, ctx)
     if e.score >= 1 then
       e.matches = match.matches
       e.exact = e:get_filter_text() == inputs[o] or e:get_word() == inputs[o]
-      table.insert(entries, e)
+
+      local filter = self:get_source_config().filter
+      if not filter or not filter(e, ctx) then
```

Comment:
```suggestion
      if type(filter) ~= 'function' or filter(e, ctx) then
```

if no filter was defined, insert
otherwise, if the filter was defined and it passes for the entry/context pair, insert

This could also be done by defaulting the filter to `() -> true`

Reply:
>This could also be done by defaulting the filter to `() -> true`

+1 to this approach! I find setting a default filter of `() -> true` and then always applying the filter easier to understand. [Here's a talk on this approach](https://www.youtube.com/watch?v=AnZ0uTOerUI) by Michael Feathers.

Reply:
I do like that approach better. I'd implement it, but I got the feeling that @hrsh7th cooking up something on his own.

Reply:
@hrsh7th seems not active recently.
But it would be better to have a discuss with he to avoid you are working on the same feature.

Reply:
See: [03a7a91](https://github.com/hrsh7th/nvim-cmp/pull/1067/commits/03a7a91c7e1da6ce0147942a91b08fe3c52884d9)